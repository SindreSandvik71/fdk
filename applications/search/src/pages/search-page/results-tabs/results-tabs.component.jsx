import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FeatureToggle } from 'react-feature-toggles';

import { addOrReplaceParamWithoutURL } from '../../../lib/addOrReplaceUrlParam';
import localization from '../../../lib/localization';
import './results-tabs.scss';
import { FEATURES } from '../../../app/features';

export const ResultsTabs = props => {
  const { location, countDatasets, countTerms } = props;
  let { search } = location;
  search = addOrReplaceParamWithoutURL(search, 'from', '');
  return (
    <div className="row">
      <div className="col-12 col-lg-8 offset-lg-4">
        <ul className="search-results-tabs">
          <li className={location.pathname === '/' ? 'li-active' : ''}>
            <Link
              to={{ pathname: '/', search }}
              aria-label="Link til side for datasett:"
            >
              {localization.page.datasetTab}
              <span>&nbsp;({countDatasets})</span>
            </Link>
          </li>
          <FeatureToggle featureName={FEATURES.API}>
            <li className={location.pathname === '/api' ? 'li-active' : ''}>
              <Link
                to={{ pathname: '/api', search }}
                aria-label="Link til side for api:"
              >
                {localization.page.apiTab}
                <span>&nbsp;(beta)</span>
              </Link>
            </li>
          </FeatureToggle>
          <li className={location.pathname === '/concepts' ? 'li-active' : ''}>
            <Link
              to={{ pathname: '/concepts', search }}
              aria-label="Link til side for begrep:"
            >
              {localization.page.termTab}
              <span>&nbsp;({countTerms})</span>
            </Link>
          </li>
        </ul>
      </div>
    </div>
  );
};

ResultsTabs.defaultProps = {
  countDatasets: null,
  countTerms: null
};

ResultsTabs.propTypes = {
  location: PropTypes.object.isRequired,
  countDatasets: PropTypes.number,
  countTerms: PropTypes.number
};
