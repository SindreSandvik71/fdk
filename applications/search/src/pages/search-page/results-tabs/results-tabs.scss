@import '../../../../node_modules/designsystemet/fdk-designsystem-bootstrap3/scss/colors.scss';

.search-results-tabs {
  margin-bottom: 0;
  padding-left: 0;
  li {
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    border-color: $color3;
    vertical-align: bottom;
    float: none;
    display: inline-block;
    margin-right: 1em;
    line-height: 3em;
    button, a {
      background: none;
      border: 0;
      box-sizing: border-box;
      color: $color1;
      padding: .8em 1.5em;
      font-weight: 500;
      text-decoration: none;
    }
  }

  .li-active {
    background-color: $color6;
    font-size: 1.8rem;
  }
}

/* Medium Devices, Desktops */
@media only screen and (min-width : 992px) {
  .search-results-tabs {
    li {
      width: 14em;
      button, a {
        padding: 1em 4em;
      }
    }

    .li-active {
      font-size: 1.8rem;
    }
  }
}

/* Large Devices, Wide Screens */
@media only screen and (min-width : 1200px) {

}