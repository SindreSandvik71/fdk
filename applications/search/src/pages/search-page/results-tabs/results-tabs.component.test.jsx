import React from 'react';
import { shallow } from 'enzyme';
import { ResultsTabs } from './results-tabs.component';

test('should render ResultsTabs with "Dataset" active correctly', () => {
  const defaultProps = {
    location: {
      pathname: '/',
      search: '',
      hash: '',
      key: 'ublg0e'
    }
  };
  const wrapper = shallow(<ResultsTabs {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ResultsTabs with "Concepts" active correctly', () => {
  const defaultProps = {
    location: {
      pathname: '/concepts',
      search: '',
      hash: '',
      key: 'ublg0e'
    }
  };
  const wrapper = shallow(<ResultsTabs {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
