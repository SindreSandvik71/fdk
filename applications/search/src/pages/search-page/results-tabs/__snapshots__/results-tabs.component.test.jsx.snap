// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ResultsTabs with "Concepts" active correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-lg-8 offset-lg-4"
  >
    <ul
      className="search-results-tabs"
    >
      <li
        className=""
      >
        <Link
          aria-label="Link til side for datasett:"
          replace={false}
          to={
            Object {
              "pathname": "/",
              "search": "?",
            }
          }
        >
          Dataset
          <span>
             (
            )
          </span>
        </Link>
      </li>
      <FeatureToggle
        featureName="API"
        showOnlyWhenDisabled={false}
      >
        <li
          className=""
        >
          <Link
            aria-label="Link til side for api:"
            replace={false}
            to={
              Object {
                "pathname": "/apis",
                "search": "?",
              }
            }
          >
            Api
            <span>
               (beta)
            </span>
          </Link>
        </li>
      </FeatureToggle>
      <li
        className="li-active"
      >
        <Link
          aria-label="Link til side for begrep:"
          replace={false}
          to={
            Object {
              "pathname": "/concepts",
              "search": "?",
            }
          }
        >
          Concepts
          <span>
             (
            )
          </span>
        </Link>
      </li>
    </ul>
  </div>
</div>
`;

exports[`should render ResultsTabs with "Dataset" active correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-lg-8 offset-lg-4"
  >
    <ul
      className="search-results-tabs"
    >
      <li
        className="li-active"
      >
        <Link
          aria-label="Link til side for datasett:"
          replace={false}
          to={
            Object {
              "pathname": "/",
              "search": "?",
            }
          }
        >
          Dataset
          <span>
             (
            )
          </span>
        </Link>
      </li>
      <FeatureToggle
        featureName="API"
        showOnlyWhenDisabled={false}
      >
        <li
          className=""
        >
          <Link
            aria-label="Link til side for api:"
            replace={false}
            to={
              Object {
                "pathname": "/apis",
                "search": "?",
              }
            }
          >
            Api
            <span>
               (beta)
            </span>
          </Link>
        </li>
      </FeatureToggle>
      <li
        className=""
      >
        <Link
          aria-label="Link til side for begrep:"
          replace={false}
          to={
            Object {
              "pathname": "/concepts",
              "search": "?",
            }
          }
        >
          Concepts
          <span>
             (
            )
          </span>
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
